lms/djangoapps/lti_provider/tests/test_management.py

```diff
@@ -0,0 +1,87 @@
+from xmodule.modulestore.xml_importer import import_course_from_xml
+
+from lti_provider.management.commands import rescore_lti
+from lti_provider.models import GradedAssignment, LtiConsumer, OutcomeService
+
+class CommandArgsTestCase(TestCase):
+    """
+    Tests management commands parse arguments properly
+    """
+    def test_command_arguments(self):
+        cmd = rescore_lti.Command()
+        parser = cmd.create_parser('./manage.py', 'rescore_lti')
+        args = parser.parse_args(['course-v1:edX+test_course+2525_fall', 'UBC/Law281/2015_T1'])
+        self.assertEqual(len(args.course_keys), 2)
+        key = args.course_keys[0]
+        self.assertIsInstance(key, CourseKey)
+        self.assertEqual(unicode(key), 'course-v1:edX+test_course+2525_fall')
+
+
+class RescoreLTITestCase(SharedModuleStoreTestCase):
+    """
+    Test manage.py rescore_lti command
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(RescoreLTITestCase, cls).setUpClass()
+        cls.course_key = cls.store.make_course_key('edX', 'lti_provider', '3000')
+        course_items = import_course_from_xml(
+            cls.store,
+            'test_user',
+            TEST_DATA_DIR,
+            source_dirs=['simple'],
+            static_content_store=None,
+            target_id=cls.course_key,
+            raise_on_failure=True,
+            create_if_not_present=True,
+        )
+        cls.course_items = course_items
+        cls.course = course_items[0]
+        print dir(cls.course)
+
+    def setUp(self):
+        super(RescoreLTITestCase, self).setUp()
+        self.user = UserFactory()
+        self.user2 = UserFactory(username='anotheruser')
+        self.client.login(username=self.user.username, password="test")
+
+    def _configure_lti(self, usage_key):
+        consumer = LtiConsumer.objects.create()
+        outcome_service = OutcomeService.objects.create(lti_consumer=consumer, lis_outcome_service_url='https://lol.tools')
+        with patch('lti_provider.outcomes.send_score_update'):  # Avoid running this.
+            self.assignments = [
+                GradedAssignment.objects.create(
+                    user=self.user,
+                    course_key=self.course_key,
+                    usage_key=usage_key,
+                    outcome_service=outcome_service,
+                    lis_result_sourcedid='abc',
+                ),
+                GradedAssignment.objects.create(
+                    user=self.user2,
+                    course_key=self.course_key,
+                    usage_key=usage_key,
+                    outcome_service=outcome_service,
+                    lis_result_sourcedid='xyz',
+                ),
+            ]
+
+    def test_rescore_lti_enqueues_tasks(self):
+        cmd = rescore_lti.Command()
+        self._configure_lti(UsageKey.from_string('block-v1:edX+lti_provider+3000+type@chapter+block@chapter_2'))
+        with patch('lti_provider.tasks.send_composite_outcome.apply_async') as mock_task:
```

Comment:
Consider mocking the call to `lti_provider.outcomes.send_score_update` instead - to have more guarantees of the success of the call - rather than just making sure the management command calls `send_composite_outcome`.